### PR TITLE
DevEx - refactoring the app context into multiple files

### PR DESCRIPTION
**Pull request by codyseibert**
- the application context was too many lines which made saving the files and editing a pain, splitting up the useCases, useCaseHelpers, and persistenceGateway into separate files.
- include some additional persistence methods inside the validation decorate

### CODE REVIEW

File: web-api/src/getPersistenceGateway.js [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+    createJobStatus,
+    createMessage,
+    createNewPetitionerUser,
+    createNewPractitionerUser,
+    createOrUpdatePractitionerUser,
+    createOrUpdateUser,
+    createTrialSession,
+    createTrialSessionWorkingCopy,
+    deleteKeyCount,
+    fetchPendingItems,
```
Comment (rosiehsmith) [inline web-api/src/getPersistenceGateway.js:498]: this one is also in `getuseCaseHelpers` - is that necessary?

Comment (rosiehsmith) [reply web-api/src/getPersistenceGateway.js:498]: by this one i mean `fetchPendingItems`

**Review by rosiehsmith — APPROVED** [review]
🎉 nice refactoring

File: web-api/src/getPersistenceGateway.js (hunk 1 above)
Comment (codyseibert) [reply web-api/src/getPersistenceGateway.js:498]: good catch, I think I had some extra persistence files in the wrong file

**Pull request merged**